### ISSUE TITLE
Hwloc 2.1 requires C99, which MSVC doesn't fully support

### ISSUE THREAD

**Issue by matthiasdiener** [UIUC-PPL/charm#2606]
Building the master branch on Windows with MSVC 2015 (cl.exe 19.0) results in the following error:

```
configure:16844: checking for /home/mdiener/tmp.rteXRk1XUT/netlrts-win-x86_64/bin/unix2nt_cc option to accept ISO C99
configure:16993: /home/mdiener/tmp.rteXRk1XUT/netlrts-win-x86_64/bin/unix2nt_cc  -c -D_CRT_SECURE_NO_DEPRECATE   conftest.c >&5
conftest.c
conftest.c(202): error C2146: syntax error: missing ')' before identifier 'text'
conftest.c(202): error C2061: syntax error: identifier 'text'
conftest.c(202): error C2059: syntax error: ';'
conftest.c(202): error C2059: syntax error: ')'
conftest.c(203): error C2449: found '{' at file scope (missing function header?)
conftest.c(210): error C2059: syntax error: '}'
conftest.c(255): error C2146: syntax error: missing ';' before identifier 'newvar'
conftest.c(255): error C2065: 'newvar': undeclared identifier
conftest.c(255): warning C4047: '=': 'int' differs in levels of indirection from 'char [15]'
conftest.c(277): error C2057: expected constant expression
conftest.c(277): error C2466: cannot allocate an array of constant size 0
conftest.c(277): error C2133: 'dynamic_array': unknown size
conftest.c(281): error C2065: 'newvar': undeclared identifier
conftest.c(281): error C2109: subscript requires array or pointer type
unix2nt_cc Fatal Error in directory /home/mdiener/tmp.rteXRk1XUT/netlrts-win-x86_64/tmp
   Error executing CL.EXE -nologo -W1 -EHsc -D_WINDOWS -DNOMINMAX -MT -c -D_CRT_SECURE_NO_DEPRECATE -Tcconftest.c
unix2nt_cc exiting...
```

**Comment by evan-charmworks** [comment]
hwloc 2.1 added this configure test. https://github.com/open-mpi/hwloc/commit/f2226f76e104923a76c5d09328284104abad6b01

Since it calls an autotools builtin, `AC_PROG_CC_C99`, it is entirely possible that hwloc builds correctly and it is only this test that is failing.

**Comment by matthiasdiener** [comment]
Workaround has been merged (#2614).
